packages/app/prisma-utils/src/prismaTransaction.ts

```diff
@@ -58,9 +60,12 @@ export const prismaTransaction = (async <T, P extends PrismaClient>(
 		}
 
 		result = await executeTransactionTry(prisma, arg, options)
-		if (result.result || !isRetryAllowed(result, optionsWithDefaults.DbDriver)) {
-			break
-		}
+		if (result.result) break
+
+		const retryAllowed = isRetryAllowed(result, optionsWithDefaults.DbDriver)
+		if (!retryAllowed) break
+
+		if (retryAllowed === 'increase-timeout') optionsWithDefaults.timeout *= 2 // TODO: implement smart timeout increase
```

Comment:
This incremental behavior is only an example, if we agree this is a good feature we should define how the increase should look like